### DISCUSSION TITLE
Giving Octokit .NET & the community the ❤️  it deserves

### DISCUSSION THREAD

**Discussion by nickfloyd** [octokit/octokit.net#2499] [Announcements]
### The Past

🕚 To say it's "been a while" is a bit of an understatement 😬.  Octokit.NET started back in April 2012 (initially named Burr with the first [commit](https://github.com/octokit/octokit.net/commit/0cbf4ab360bd6a2aa172566bdd1167133c0dfaac) by @tclem). 

I remember being a user of Octopi (2013) and seeing the tremendous effort that was being put in and the value of having a .NET GitHub SDK.  Given that you are reading this, I imagine you have your own Octokit story (past or present) - in some way this SDK has impacted you.

Whether you’ve been using Octokit.NET for nearly a decade or you’ve just found this project today, there is one thing I think we can all agree on:  because of the efforts of folks like @shiftkey, @haacked, @ryangribble, @tclem, @half-ogre (and so many more) our community has been able to build incredible things; dev lives were made better as a result of a community making time to make this SDK.

Even given all of this effort, this library has sat unchanged/unmaintained for a long time - despite the need for this SDK and community desire which were still there. This happens in open source sometimes: "day jobs" get in the way, contributors move to other projects, and other things come up - one thing I love about our community here is that we were still hungry 🍖 enough to not let it go.

----

### The Present

GitHub as an organization has decided to invest in this project and the community behind it:  we have started to build an SDK team to own the [Octokit SDKs and source](https://github.com/octokit) (spanning across 66 different repositories) 🎉!

You've probably already noticed but we've started to engage with the community, [ship changes](https://github.com/octokit/octokit.net/releases), and work on developer experience for this and our other SDKs in [Ruby](https://github.com/octokit/octokit.rb) and [JavaScript](https://github.com/octokit/octokit.js).  

✅  **Goal 1: We're starting with our current goal to get this SDK and the other ones to a 100% shippable state.** 

100% shippable means:

- The SDK can be versioned and released to its corresponding platform (Ruby gems, Nuget, NPM).
- The SDK has its release process documented and is repeatable.
- The SDK’s release is pipelined and CI is implemented and working.
- The source code for the SDK is covered by dependabot and code scanning.
- The SDK documentation reflects reality.
- All security and dependency vulnerabilities have been fixed.
- A version of the given SDK has been released with the above completed. 

➖ **Goal 2: Ship value, remove roadblocks, and enable the community.**

This means that:
- We are going through the 100's of PRs and Issues and determining what can be shipped and shipping it.
- Connecting with the community and having discussions on reality-based needs and pain points.
- We are fixing bugs and sanding out rough spots in the SDK.
- Cleaning up any legacy patterns and implementations (such as REST API previews)
- We are implementing patterns, practices, and processes to reduce the friction of getting things done and shipped.
- We are adding (and removing) tooling and workflows that will better equip our community

----

### The Future

The team at GitHub will get to work on Octokit.NET and our other SDKs as their full-time jobs, but we need involvement from the community. All of your contributions and engagement provide a massive value to the present and future of the SDKs; we want/need you to help to shape our roadmap!

We've got what we think is a solid roadmap to enable the next generation of things but we need your thoughts and input.  Some of the things we see coming down the road are:

- Making it as easy and fast as possible for open source contributors to make changes to the SDK and get them merged
- Generating models and clients based on the [OpenAPI descriptions](https://github.com/github/rest-api-description), so that we can all put more energy into the more innovative features of the SDK rather than boilerplate
- Fully automating the CI pipeline to ensure that the changes that are being made are stable and in the hands of the community quickly
- Potentially abstracting generated models so that they can be used in any disparate libraries (as well as the SDK) and be versioned.
- Create a plugin architecture for being able to introduce recipes that provide common workflows
- more...

----

Right now we are looking to grow the GitHub SDK team! The SDKs are a priority and our community’s needs are important to us, which is why we are so stoked about the potential energy stored in the current, new, and future creators of these SDKs! 💥 

❤️  We hope you'll join us on this journey to make it easy and enjoyable for .NET developers to build great things with these SDKs. Please reach out to me (@nickfloyd) or @timrogers if you have any questions and/or want to get involved!

**Comment by haacked** [comment]
This is exciting to hear!

![](https://media.giphy.com/media/5mBE2MiMVFITS/giphy.gif)

**Reply by JonruAlveus** [reply]
100% agreed! Let’s make this Repo the best it can be!

**Comment by nerocui** [comment]
YAY! My app depends heavily on the .NET SDK, and I'm planning to move to V4. This is great news!

**Comment by Lamparter** [comment]
So excited for this to be back-in-business!

**Comment by Brend-Smits** [comment]
Exciting news. I will definitely be keeping an eye out for the different Octokit projects from now on. Keep it up you all.

**Comment by 0x5bfa** [comment]
Yey!! So excited to hear!!

**Comment by JonruAlveus** [comment]
Really excited about this, but I also want to thank the GitHub team for picking these repos up and all the effort they are putting in right now and I suspect the huge amount coming up (and I apologise in advance for the amount of work I’m likely to make you do/endless long questions for you to answer😄)! This will definitely be a team effort in collaboration and I’m really glad to be part of it.

**Comment by salty-flower** [comment]
Is .NET 8 migration on the roadmap?

**Reply by Lamparter** [reply]
Yes

**Comment by Dan-Albrecht** [comment]
Has the love worn out? The trivial dependabot PRs aren't getting merged and there's been no commits since the beginning of the year.

**Reply by Lamparter** [reply]
Unfortunately seems like it

**Comment by kirsan31** [comment]
It seems that the the project is dead :( **Critical** bugs  (https://github.com/octokit/octokit.net/issues/2959) from 2024 is not fixed but closed as of inactivity 😭
